### DISCUSSION TITLE
set default page title to its filename?

### DISCUSSION THREAD

**Discussion by writeblankspace** [lumeland/lume#646] [Q&A] [answered]
```ts
export const layout = "layouts/blogPage.vto";
export const category = "blog";

export const title = (page: any) => {
	return page.data.filename;
};
```

I tried this in `_data.ts` where my `_data.json` used to be, but the title in my page became `(page)=>{ return page.data.filename; }` which is really weird. My `layout` and `category` are working fine.

Am I doing this wrong?

**Comment by oscarotero** [comment, marked as answer]
That's because `title` is not a string, but a function, so if you print it in your templates, it prints the function body ([fn.toString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/toString)).
If you want to set default values, I recommend to use a preprocessor. For example:


```js
site.preprocess([".html"], (pages) => {
  for (const page of pages) {
    page.data.title = page.data.basename;
  }
});
```

In your specific case, you can use the `basename` variable, that contains the filename after removing the extension. In the layout, you can do:

```html
<h1>{{ title || basename }}</h1>
```

**Reply by writeblankspace** [reply]
Thank you; this worked.

I did have to add an `if` statement to prevent *all* pages' names changing though.
